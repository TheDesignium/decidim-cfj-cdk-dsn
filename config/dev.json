{
  "rds": {
    "snapshot": true,
    "snapshotIdentifier": "decidim-master-2022-11-07",
    "instanceType": "t2.micro",
    "deletionProtection": false,
    "allocatedStorage": 20
  },

  "aws": {
    "accountId": "887442827229",
    "region": "ap-northeast-1"
  },

  "cacheNodeType" : "cache.t2.micro",
  "engineVersion" : "6.x",
  "numCacheNodes" : 1,
  "automaticFailoverEnabled": false,

  "smtpDomain": "diycities.jp",
  "domain": "diycities.jp",
  "repository": "decidim-cfj",
  "tag": "staging-80373039",
  "certificates": [
    "arn:aws:acm:ap-northeast-1:887442827229:certificate/faa7a120-3843-4561-93ec-c3d313356ad5",
    "arn:aws:acm:ap-northeast-1:887442827229:certificate/b02c7415-d6f8-4794-994e-fe5689ca7b74"
  ],
  "cloudfrontCertificate": "arn:aws:acm:us-east-1:887442827229:certificate/8aec1d57-e068-47f3-a180-e0c1fbb782de"
}
