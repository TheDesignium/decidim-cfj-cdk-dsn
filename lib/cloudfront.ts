import {
  aws_certificatemanager,
  aws_cloudfront as cloudfront,
  aws_cloudfront_origins,
  aws_wafv2,
  Stack
} from "aws-cdk-lib";
import { Construct } from "constructs";
import { BaseStackProps } from "./props";
import { AllowedMethods, CachePolicy, OriginRequestPolicy, ViewerProtocolPolicy } from "aws-cdk-lib/aws-cloudfront";

export interface CloudFrontProps extends BaseStackProps {
  certificateArn: string
  domain: string
}

export class CloudFrontStack extends Stack {
  constructor(scope: Construct, id: string, props: CloudFrontProps) {
    super(scope, id, props);
    const endpoint = `${ props.stage }-${ props.serviceName }-alb-origin.${ props.domain }`;
    const origin = new aws_cloudfront_origins.HttpOrigin(endpoint)
    origin.bind(this, { originId: "defaultEndPoint" })

    const waf = new aws_wafv2.CfnWebACL(this, "CfnWebACL", {
        name: `${ props.stage }-${ props.serviceName }-webAcl`,
        defaultAction: { allow: {} },
        scope: "CLOUDFRONT",
        visibilityConfig: {
          cloudWatchMetricsEnabled: true,
          sampledRequestsEnabled: true,
          metricName: `${ props.stage }-${ props.serviceName }-webAcl-metrics`
        },
        description: `Web ACL for ${ props.stage }-${ props.serviceName }-cloudfront`,
        rules: [
          {
            name: `${ props.stage }-${ props.serviceName }-AWSManagedRulesCommonRuleSet`,
            priority: 1,
            statement: {
              managedRuleGroupStatement: {
                vendorName: "AWS",
                name: "AWSManagedRulesCommonRuleSet",
              },
            },
            overrideAction: { none: {} },
            visibilityConfig: {
              cloudWatchMetricsEnabled: true,
              sampledRequestsEnabled: true,
              metricName: `${ props.stage }-${ props.serviceName }-AWSManagedRulesCommonRuleSet`,
            },
          },
          {
            name: `${ props.stage }-${ props.serviceName }-AWSManagedRulesKnownBadInputsRuleSet`,
            priority: 2,
            statement: {
              managedRuleGroupStatement: {
                vendorName: "AWS",
                name: "AWSManagedRulesKnownBadInputsRuleSet",
              },
            },
            overrideAction: { none: {} },
            visibilityConfig: {
              cloudWatchMetricsEnabled: true,
              sampledRequestsEnabled: true,
              metricName: `${ props.stage }-${ props.serviceName }-AWSManagedRulesKnownBadInputsRuleSet`,
            },
          },
          {
            name: `${ props.stage }-${ props.serviceName }-AWSManagedRulesAmazonIpReputationList`,
            priority: 3,
            statement: {
              managedRuleGroupStatement: {
                vendorName: "AWS",
                name: "AWSManagedRulesAmazonIpReputationList",
              },
            },
            overrideAction: { none: {} },
            visibilityConfig: {
              cloudWatchMetricsEnabled: true,
              sampledRequestsEnabled: true,
              metricName: `${ props.stage }-${ props.serviceName }-AWSManagedRulesAmazonIpReputationList`,
            },
          },
          {
            name: `${ props.stage }-${ props.serviceName }-AWSManagedRulesLinuxRuleSet`,
            priority: 4,
            statement: {
              managedRuleGroupStatement: {
                vendorName: "AWS",
                name: "AWSManagedRulesLinuxRuleSet",
              },
            },
            overrideAction: { none: {} },
            visibilityConfig: {
              cloudWatchMetricsEnabled: true,
              sampledRequestsEnabled: true,
              metricName: `${ props.stage }-${ props.serviceName }-AWSManagedRulesLinuxRuleSet`,
            },
          },
          {
            name: `${ props.stage }-${ props.serviceName }-AWSManagedRulesSQLiRuleSet`,
            priority: 5,
            statement: {
              managedRuleGroupStatement: {
                vendorName: "AWS",
                name: "AWSManagedRulesSQLiRuleSet",
              },
            },
            overrideAction: { none: {} },
            visibilityConfig: {
              cloudWatchMetricsEnabled: true,
              sampledRequestsEnabled: true,
              metricName: `${ props.stage }-${ props.serviceName }-AWSManagedRulesSQLiRuleSet`,
            },
          },
          {
            name: `${ props.stage }-${ props.serviceName }-AllowSystemLogin`,
            priority: 6,
            statement: {
              andStatement: {
                statements: [{
                  byteMatchStatement: {
                    searchString: `${ props.stage }-${ props.serviceName }-alb-origin`,
                    fieldToMatch: {
                      singleHeader: {
                        name: 'host'
                      }
                    },
                    textTransformations: [
                      {
                        priority: 0,
                        type: 'LOWERCASE'
                      }
                    ],
                    positionalConstraint: "EXACTLY"
                  }
                },
                  {
                    byteMatchStatement: {
                      searchString: "system/admins/sign_in",
                      fieldToMatch: {
                        uriPath: {}
                      },
                      textTransformations: [
                        {
                          priority: 0,
                          type: 'LOWERCASE'
                        }
                      ],
                      positionalConstraint: "CONTAINS"
                    }
                  }
                ]
              }
            },
            action: {
              allow: {}
            },
            visibilityConfig: {
              cloudWatchMetricsEnabled: true,
              sampledRequestsEnabled: true,
              metricName: `${ props.stage }-${ props.serviceName }-AllowSystemLogin`,
            },
          },
          {
            name: `${ props.stage }-${ props.serviceName }-SystemLoginBlock`,
            priority: 7,
            statement: {
              byteMatchStatement: {
                searchString: 'system/admins/sign_in',
                fieldToMatch: {
                  uriPath: {}
                },
                textTransformations: [
                  {
                    priority: 0,
                    type: "LOWERCASE"
                  }
                ],
                positionalConstraint: "CONTAINS"
              }
            },

            action: {
              block: {
                customResponse: {
                  responseCode: 403,
                  customResponseBodyKey: "disable-action"
                }
              }
            },
            visibilityConfig: {
              cloudWatchMetricsEnabled: true,
              sampledRequestsEnabled: true,
              metricName: `${ props.stage }-${ props.serviceName }-AllowSystemLogin`,
            },
          }
        ],
        customResponseBodies: {
          'disable-action': {
            content: '<div>error: access denied</div>',
            contentType: 'TEXT_HTML'
          }
        }
      }
    )

    if (props.stage === "prd-v0252") {
      new cloudfront.Distribution(this, 'Distribution', {
        priceClass: cloudfront.PriceClass.PRICE_CLASS_ALL,
        defaultBehavior: {
          origin: origin,
          allowedMethods: AllowedMethods.ALLOW_ALL,
          viewerProtocolPolicy: ViewerProtocolPolicy.REDIRECT_TO_HTTPS,
          cachePolicy: CachePolicy.CACHING_DISABLED,
          originRequestPolicy: OriginRequestPolicy.ALL_VIEWER
        },
        comment: `${ props.stage }-${ props.serviceName }-cloudfront`,
        domainNames: [endpoint, `*.${ props.domain }`],
        certificate: aws_certificatemanager.Certificate.fromCertificateArn(this, 'cloudFrontCertificate', props.certificateArn),
        webAclId: waf.attrArn
      }).addBehavior('decidim-packs/*', origin, {
        allowedMethods: AllowedMethods.ALLOW_GET_HEAD_OPTIONS,
        viewerProtocolPolicy: ViewerProtocolPolicy.REDIRECT_TO_HTTPS,
        cachePolicy: CachePolicy.CACHING_OPTIMIZED
      })
    } else {
      new cloudfront.Distribution(this, 'Distribution', {
        priceClass: cloudfront.PriceClass.PRICE_CLASS_ALL,
        defaultBehavior: {
          origin: origin,
          allowedMethods: AllowedMethods.ALLOW_ALL,
          viewerProtocolPolicy: ViewerProtocolPolicy.REDIRECT_TO_HTTPS,
          cachePolicy: CachePolicy.CACHING_DISABLED,
          originRequestPolicy: OriginRequestPolicy.ALL_VIEWER
        },
        comment: `${ props.stage }-${ props.serviceName }-cloudfront`,
        domainNames: [endpoint],
        certificate: aws_certificatemanager.Certificate.fromCertificateArn(this, 'cloudFrontCertificate', props.certificateArn),
        webAclId: waf.attrArn
      }).addBehavior('decidim-packs/*', origin, {
        allowedMethods: AllowedMethods.ALLOW_GET_HEAD_OPTIONS,
        viewerProtocolPolicy: ViewerProtocolPolicy.REDIRECT_TO_HTTPS,
        cachePolicy: CachePolicy.CACHING_OPTIMIZED
      })
    }
  }
}
