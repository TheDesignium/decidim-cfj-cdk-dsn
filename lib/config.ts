import { InstanceType } from "aws-cdk-lib/aws-ec2";
import { PostgresEngineVersion } from "aws-cdk-lib/aws-rds";

export interface SubnetConfig {
  subnetId: string;
  availabilityZone: string;
  routeTableId: string;
}

export interface VpcConfig {
  vpcId: string;
  cidrBlock: string;
  availabilityZones: string[];
  publicSubnets: SubnetConfig[];
  privateSubnets: SubnetConfig[];
}

export interface RdsConfig {
  postgresVersion: PostgresEngineVersion
  snapshot: boolean
  snapshotIdentifier: string
  instanceType: InstanceType
  deletionProtection: boolean
  allocatedStorage: number
  multiAz: boolean
}

export interface Config {
  stage: string

  vpc?: VpcConfig;
  aws: {
    accountId: string;
    region: string;
  };

  // s3
  versioned: boolean

  // rds
  rds: RdsConfig

  // elastiCache
  cacheNodeType: string
  engineVersion: string
  numCacheNodes: number
  automaticFailoverEnabled: boolean

  // service
  smtpDomain: string,
  domain: string,
  repository: string
  tag: string
  certificates: string[]
}

export function getConfig (stage: string): Config {
  return require(`../config/${ stage }.json`)
}