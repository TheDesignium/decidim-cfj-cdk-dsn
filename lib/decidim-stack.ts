import * as cdk from 'aws-cdk-lib';
import {
  aws_ec2, aws_ecr,
  aws_ecs as ecs,
  aws_elasticloadbalancingv2 as elbv2,
  aws_logs as logs, aws_s3,
  aws_ssm as ssm,
  Duration,
  RemovalPolicy
} from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { BaseStackProps } from "./props";
import { RetentionDays } from "aws-cdk-lib/aws-logs";
import { ApplicationTargetGroup, ListenerCertificate } from "aws-cdk-lib/aws-elasticloadbalancingv2";

export interface DecidimStackProps extends BaseStackProps {
  vpc: aws_ec2.IVpc
  certificate: ListenerCertificate[]
  securityGroup: aws_ec2.SecurityGroup
  securityGroupForAlb: aws_ec2.SecurityGroup
  containerSpec?: {
    cpu: number;
    memoryLimitMiB: number;
  }
  bucketName: string
  domain: string
  repository: string
  rds: string
  cache: string
}

export class DecidimStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props: DecidimStackProps) {
    super(scope, id, props);

    // ECS Cluster
    const cluster = new ecs.Cluster(this, 'DecidimCluster', {
      vpc: props.vpc,
      clusterName: `${ props.stage }DecidimCluster`
    })

    // Task Definition
    const taskDefinition = new ecs.FargateTaskDefinition(
      this,
      "decidimTaskDefinition",
      {
        cpu: props.containerSpec ? props.containerSpec.cpu : 1024,
        memoryLimitMiB: props.containerSpec
          ? props.containerSpec?.memoryLimitMiB
          : 2048,
        family: `${ props.stage }DecidimTaskDefinition`,
      }
    );

    const DecidimContainerEnvironment: { [key: string]: string } = {
      AWS_ACCESS_KEY_ID: ssm.StringParameter.valueForTypedStringParameter(this, `/decidim-cfj/${props.stage}/AWS_ACCESS_KEY_ID`),
      AWS_SECRET_ACCESS_KEY: ssm.StringParameter.valueForTypedStringParameter(this, `/decidim-cfj/${props.stage}/AWS_SECRET_ACCESS_KEY`),
      AWS_CLOUD_FRONT_END_POINT: ssm.StringParameter.valueForTypedStringParameter(this, `/decidim-cfj/${props.stage}/AWS_CLOUD_FRONT_END_POINT`),
      REDIS_URL: `redis://${props.cache}:6379`,
      RDS_DB_NAME: ssm.StringParameter.valueForTypedStringParameter(this, `/decidim-cfj/${props.stage}/RDS_DB_NAME`),
      RDS_HOSTNAME: props.rds,
      RDS_USERNAME: ssm.StringParameter.valueForTypedStringParameter(this, `/decidim-cfj/${props.stage}/RDS_USERNAME`),
      RDS_PASSWORD: ssm.StringParameter.valueForTypedStringParameter(this, `/decidim-cfj/${props.stage}/RDS_PASSWORD`),
      SECRET_KEY_BASE: ssm.StringParameter.valueForTypedStringParameter(this, `/decidim-cfj/${props.stage}/SECRET_KEY_BASE`),
      NEW_RELIC_LICENSE_KEY: ssm.StringParameter.valueForTypedStringParameter(this, `/decidim-cfj/${props.stage}/NEW_RELIC_LICENSE_KEY`),
      SMTP_ADDRESS: ssm.StringParameter.valueForTypedStringParameter(this, `/decidim-cfj/${props.stage}/SMTP_ADDRESS`),
      SMTP_USERNAME: ssm.StringParameter.valueForTypedStringParameter(this, `/decidim-cfj/${props.stage}/SMTP_USERNAME`),
      SMTP_PASSWORD: ssm.StringParameter.valueForTypedStringParameter(this, `/decidim-cfj/${props.stage}/SMTP_PASSWORD`),
      SMTP_DOMAIN: props.domain,
      AWS_BUCKET_NAME: props.bucketName,
      DECIDIM_COMMENTS_LIMIT: "30",
    };

    const decidimRepository = new aws_ecr.Repository(this, 'DecidimRepository', {
      repositoryName: props.repository
    })

    const container = taskDefinition.addContainer('appContainer', {
      image: ecs.ContainerImage.fromEcrRepository(decidimRepository),
      environment: DecidimContainerEnvironment,
      logging: ecs.LogDriver.awsLogs({
        logGroup: new logs.LogGroup(this, 'DecidimLogGroup', {
          logGroupName: `${ props.stage }-${ props.serviceName }-serviceLogGroup`,
          removalPolicy: RemovalPolicy.DESTROY,
          retention: RetentionDays.TWO_WEEKS
        }),
        streamPrefix: 'app'
      }),
      healthCheck: {
        command: [
          'CMD-SHELL',
          `curl --fail -s http://localhost:3000 || exit 1`
        ],
        retries: 3,
        startPeriod: Duration.minutes(2),
        interval: Duration.minutes(1),
      },
    })
    container.addPortMappings({
      containerPort: 3000
    })

    // taskDefinition.addContainer('sidekiqContainer', {
    //   image: ecs.ContainerImage.fromRegistry(props.repository),
    //   environment: DecidimContainerEnvironment,
    //   logging: ecs.LogDriver.awsLogs({
    //     logGroup: new logs.LogGroup(this, 'sidekiqLogGroup', {
    //       logGroupName: `${ props.stage }-${ props.serviceName }-sidekiqLogGroup`,
    //       removalPolicy: RemovalPolicy.DESTROY,
    //       retention: RetentionDays.TWO_WEEKS
    //     }),
    //     streamPrefix: 'app'
    //   }),
    //   command: ['bundle exec sidekiq -C /app/config/sidekiq.yml'],
    //   healthCheck: {
    //     command: [
    //       'CMD-SHELL',
    //       `test -f sidekiq.pid`
    //     ]
    //   }
    // })

    taskDefinition.defaultContainer = container

    const ecsService = new ecs.FargateService(this, 'DecidimService', {
      cluster,
      taskDefinition,
      serviceName: `${ props.stage }DecidimService`,
      vpcSubnets: {
        subnets: props.vpc.publicSubnets
      },
      securityGroups: [props.securityGroup],
      desiredCount: 1,
      assignPublicIp: true,
      enableExecuteCommand: true // For Debug
    })
    ecsService.autoScaleTaskCount({
      minCapacity: 1,
      maxCapacity: 5
    })


    // ALB Definition
    const loadBalancer = new elbv2.ApplicationLoadBalancer(this, 'Alb', {
      vpc: props.vpc,
      internetFacing: true,
      http2Enabled: true,
      loadBalancerName: `${ props.stage }-Decidim-Alb`,
      securityGroup: props.securityGroupForAlb
    })

    const targetGroup = new ApplicationTargetGroup(this, 'TargetGroup', {
      protocol: elbv2.ApplicationProtocol.HTTP,
      port: 3000,
      healthCheck: {
        port: '3000',
        path: '/',
        protocol: elbv2.Protocol.HTTP
      },
      targets: [ecsService],
      targetGroupName: `${ props.stage }-${ props.serviceName }-TargetGroup`,
      vpc: props.vpc
    })

    loadBalancer.addListener('httpListener', {
      protocol: elbv2.ApplicationProtocol.HTTP,
      port: 80,
      defaultTargetGroups: [targetGroup]
    })

    loadBalancer.addListener('httpsListener', {
      protocol: elbv2.ApplicationProtocol.HTTP,
      port: 443,
      defaultTargetGroups: [targetGroup],
      certificates: props.certificate
    })

    // ALB Log
    const logBucket = new aws_s3.Bucket(this, `${ props.stage }AlbLogBucket`, {
      bucketName: `${ props.stage }-${ props.serviceName }-alb-logs`,
      removalPolicy: RemovalPolicy.DESTROY,
      autoDeleteObjects: true,
    })
    loadBalancer.logAccessLogs(logBucket)
  }
}
