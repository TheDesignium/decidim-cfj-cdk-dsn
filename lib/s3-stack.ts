import { aws_s3, RemovalPolicy, Stack } from "aws-cdk-lib";
import { Construct } from "constructs";
import { BaseStackProps } from "./props";

export interface S3StackProps extends BaseStackProps {
  bucketName: string
}

export class S3Stack extends Stack {
  constructor(scope: Construct, id: string, props: S3StackProps) {
    super(scope, id, props);
    
    new aws_s3.Bucket(this, 'createBucket', {
      bucketName: props.bucketName,
      versioned: props.stage === "prd",
      removalPolicy: RemovalPolicy.DESTROY
    });

  }
}