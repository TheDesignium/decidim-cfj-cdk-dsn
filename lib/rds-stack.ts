import { aws_ec2, aws_rds as rds, RemovalPolicy, Stack, aws_ssm as ssm, SecretValue } from "aws-cdk-lib";

import { Construct } from "constructs";
import { BaseStackProps } from "./props";
import {
  DatabaseInstance,
  DatabaseInstanceEngine,
  DatabaseInstanceSourceProps,
  StorageType
} from "aws-cdk-lib/aws-rds";
import { RdsConfig } from "./config";

export interface RdsStackProps extends BaseStackProps {
  rds: RdsConfig
  vpc: aws_ec2.IVpc
  securityGroup: aws_ec2.SecurityGroup
}

export class RdsStack extends Stack {
  public readonly rds: DatabaseInstance;

  constructor(scope: Construct, id: string, props: RdsStackProps) {
    super(scope, id, props);

    const config = props.rds;

    const rdsProps: DatabaseInstanceSourceProps = {
      engine: DatabaseInstanceEngine.postgres({ version: rds.PostgresEngineVersion.VER_12_8 }),
      instanceType: config.instanceType,
      instanceIdentifier: `${ props.stage }-${ props.serviceName }-postgresql`,
      vpc: props.vpc,
      securityGroups: [props.securityGroup],
      multiAz: config.multiAz,
      removalPolicy: RemovalPolicy.DESTROY,
      deletionProtection: config.deletionProtection,
      storageType: StorageType.GP2,
      allocatedStorage: config.allocatedStorage,
      autoMinorVersionUpgrade: true,
      deleteAutomatedBackups: false,
    }

    // snapshotから復元するかどうか
    if (config.snapshot) {
      this.rds = new rds.DatabaseInstanceFromSnapshot(this, 'restoreRds', {
        ...rdsProps,
        ...{
          snapshotIdentifier: config.snapshotIdentifier,
        }
      })
    } else {
      this.rds = new rds.DatabaseInstance(this, 'createRds', {
        ...rdsProps,
        ...{
          databaseName: ssm.StringParameter.valueForTypedStringParameter(this, `/decidim-cfj/${ props.stage }/RDS_DB_NAME`),
          credentials: {
            username: ssm.StringParameter.valueForTypedStringParameter(this, `/decidim-cfj/${ props.stage }/RDS_USERNAME`),
            password: SecretValue.unsafePlainText(ssm.StringParameter.valueForTypedStringParameter(this, `/decidim-cfj/${ props.stage }/RDS_PASSWORD`)),
          }
        }
      })
    }
  }
}
