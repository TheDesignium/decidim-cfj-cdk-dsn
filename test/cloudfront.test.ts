import * as cdk from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';

import { Config, getConfig } from "../lib/config";
import { CloudFrontStack } from "../lib/cloudfront";

test('Cloudfront Stack Created', () => {
    const app = new cdk.App();

    const stage = 'staging';
    const config: Config = getConfig(stage)
    const serviceName = `decidim`;

    const cloudfrontEnv = {
        account: config.aws.accountId,
        region: 'us-east-1'
    }

    const cloudfront = new CloudFrontStack(app, `${ stage }${ serviceName }CloudFrontStack`, {
        stage,
        serviceName,
        env: cloudfrontEnv,
        domain: config.domain,
        certificateArn: config.cloudfrontCertificate
    })

    const template = Template.fromStack(cloudfront);

    // Assert the template matches the snapshot.
    expect(template.toJSON()).toMatchSnapshot();
});

