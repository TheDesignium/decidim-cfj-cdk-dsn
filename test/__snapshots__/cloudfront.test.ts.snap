// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cloudfront Stack Created 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "Distribution830FAC52": {
      "Properties": {
        "DistributionConfig": {
          "Aliases": [
            "staging-decidim-alb-origin.diycities.jp",
          ],
          "CacheBehaviors": [
            {
              "AllowedMethods": [
                "GET",
                "HEAD",
                "OPTIONS",
              ],
              "CachePolicyId": "658327ea-f89d-4fab-a63d-7e88639e58f6",
              "Compress": true,
              "PathPattern": "decidim-packs/*",
              "TargetOriginId": "stagingdecidimCloudFrontStackDistributionOrigin1E886134A",
              "ViewerProtocolPolicy": "redirect-to-https",
            },
          ],
          "Comment": "staging-decidim-cloudfront",
          "DefaultCacheBehavior": {
            "AllowedMethods": [
              "GET",
              "HEAD",
              "OPTIONS",
              "PUT",
              "PATCH",
              "POST",
              "DELETE",
            ],
            "CachePolicyId": "4135ea2d-6df8-44a3-9df3-4b5a84be39ad",
            "Compress": true,
            "OriginRequestPolicyId": "216adef6-5c7f-47e4-b989-5492eafa07d3",
            "TargetOriginId": "stagingdecidimCloudFrontStackDistributionOrigin1E886134A",
            "ViewerProtocolPolicy": "redirect-to-https",
          },
          "Enabled": true,
          "HttpVersion": "http2",
          "IPV6Enabled": true,
          "Origins": [
            {
              "CustomOriginConfig": {
                "OriginProtocolPolicy": "https-only",
                "OriginSSLProtocols": [
                  "TLSv1.2",
                ],
              },
              "DomainName": "staging-decidim-alb-origin.diycities.jp",
              "Id": "stagingdecidimCloudFrontStackDistributionOrigin1E886134A",
            },
          ],
          "PriceClass": "PriceClass_All",
          "ViewerCertificate": {
            "AcmCertificateArn": "arn:aws:acm:us-east-1:887442827229:certificate/8aec1d57-e068-47f3-a180-e0c1fbb782de",
            "MinimumProtocolVersion": "TLSv1.2_2021",
            "SslSupportMethod": "sni-only",
          },
        },
      },
      "Type": "AWS::CloudFront::Distribution",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
